roms_tools/setup/datasets.py

```diff
@@ -174,6 +174,21 @@ def load_data(self) -> xr.Dataset:
                 "A time dimension must be available to concatenate the files."
             )
 
+        # Determine the kwargs for combining datasets
+        if re.search(r"[\*\?\[\]]", str(filename_str)) or len(matching_files) == 1:
+            # If there is a wildcard or just one file, use by_coords
+            kwargs = {"combine": "by_coords"}
+        else:
+            # Otherwise, use nested combine based on time
+            kwargs = {"combine": "nested", "concat_dim": self.dim_names["time"]}
```

Comment:
A double-nested list is another valid argument to `combine_nested` that would probably go wrong if passed here.

Reply:
But we will never create a double-nested list with the current implementation, correct?

Reply:
If a double-nested list can never get here then no problem, you're right. (I only skimmed this on the subway!)

Reply:
Haha, nice! It can only get here if the user provides a double-nested list, but we require `filename : Union[str, Path, List[Union[str, Path]]]`, which excludes that case.